sdk/go/pulumi-language-go/main.go

```diff
@@ -91,6 +92,40 @@ func (host *goLanguageHost) GetRequiredPlugins(ctx context.Context,
 	return &pulumirpc.GetRequiredPluginsResponse{}, nil
 }
 
+// findProgram attempts to find the needed program in various locations on the
+// filesystem, eventually resorting to searching in $PATH.
+func findProgram(program string) (string, error) {
+
+	// look in the same directory
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get current working directory")
+	}
+
+	cwdProgram := filepath.Join(cwd, program)
+	if _, err := os.Stat(cwdProgram); !os.IsNotExist(err) {
+		logging.V(5).Infoln("program %s found in CWD", program)
```

Comment:
Would be good to check here that what we found is actually an executable (i.e. it has the executable bit set and it's not a directory) - otherwise we're probably going to get a bad error message later on when we try to execute it.

Reply:
Though I'm not sure Go exposes the executable bit, so I'd be fine just checking that it's not a directory.